client/App.svelte

```diff
@@ -1,9 +1,25 @@
 <script>
-	import TutorialView from "./tutorial/TutorialView.svelte";
-	import HomeView from "./home/HomeView.svelte";
+	import HomeView from "./screens/HomeView.svelte";
+	import JoinView from "./screens/JoinView.svelte";
 	import GameView from "./game/GameView.svelte";
+	import { socket } from "./connectivity.js";
 
 	// can be in setup, tutorial, or playing
+
+	let roomCode = "";
+	let parts = window.location.href.split("?");
+	if (parts.length == 2) {
+		roomCode = parts[1];
+	}
+
+	socket.on("roomCreated", function(data) {
+		roomCode = data.roomCode;
+		window.history.pushState({}, "Game", "?" + data.roomCode);
```

Comment:
If you decide to use the suggestion above, this would have to change too:

```suggestion
                const queryParams = new URLSearchParams().set("code", data.roomCode);
		window.history.pushState({}, "Game", queryParams.toString());
```